feat: Implement advanced details component

pedronfigueiredo opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/4615

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

pedronfigueiredo committed:
wip

OGPoyraz commented on app/components/Views/confirmations/components/UI/info-row/info-row.styles.ts:
```diff
@@ -13,7 +14,8 @@ const styleSheet = (params: { theme: Theme }) => {
       alignItems: 'center',
       flexWrap: 'wrap',
-      paddingBottom: 8,
+      paddingBottom: 0,
+      paddingTop: isCompact ? 8 : 0,
```
Apologies, I didn't notice this change in the first round of review but this will cause an issue on other components using `InfoRow`

For example: 

![Screenshot 2025-04-23 at 13 35 12](https://github.com/user-attachments/assets/0be53708-a394-47a8-955e-ebbe61f6d79c)

Will change into 

![Screenshot 2025-04-23 at 13 35 26](https://github.com/user-attachments/assets/032c5a78-2c2a-4d18-add0-d53703466602)


We probably need this change in the `app/components/Views/confirmations/components/rows/transactions/advanced-details-row/advanced-details-row.tsx`. If so `InfoRow` accepts `style` property which will override styles of container. Shall we revert at least `paddingBottom` change and supply that wherever we need?

pedronfigueiredo committed:
wip

Daniel-Cross committed:
fix: Row Variant support in mobile (#14717)

This PR adds support for Row Variant in mobile snaps.

Fixes: https://github.com/MetaMask/snaps/issues/3183

1. Connect to the mono-repo
2. Add the following code:
```
<Row label="THIS IS A LABEL">
                <Text>THIS IS A TEXT</Text>
              </Row>
              <Row
                label="THIS IS A LABEL"
                variant="critical"
                tooltip="THIS IS A TOOLTIP"
              >
                <Text>THIS IS A TEXT</Text>
              </Row>
              <Row
                label="THIS IS A LABEL"
                variant="warning"
                tooltip="THIS IS A TOOLTIP"
              >
                <Text>THIS IS A TEXT</Text>
              </Row>
              <Row
                label="THIS IS A LABEL"
                variant="default"
                tooltip="THIS IS A TOOLTIP"
              >
                <Text>THIS IS A TEXT</Text>
              </Row>
```
3. Go to the mobile in app browser and connect to see the working row
variants

![Simulator Screenshot - iPhone 16 Pro - 2025-04-17 at 11 59
50](https://github.com/user-attachments/assets/c35f4a78-1f6f-4fc1-adc9-e525363e50f4)

- [x] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

jpuri commented on app/components/Views/confirmations/components/rows/transactions/advanced-details-row/advanced-details-row.styles.ts:
```diff
@@ -0,0 +1,19 @@
+const styleSheet = () => StyleSheet.create({
+    nonceText: {
+        textDecorationLine: 'underline',
+    },
```
Same indentation issue above.

jpuri commented on app/components/hooks/useEditNonce.test.ts:
```diff
@@ -0,0 +1,229 @@
+
+    expect(result.current.showNonceModal).toBe(false);
+  });
+}); 
```
newline issue at the last line in the file

jpuri reviewed (approved):
I added some small feedbacks to the PR, changes look good otherwise.
